core/commands/UserResetPasswordCommand.php

```diff
@@ -0,0 +1,99 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'authentication_api.php' );
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'user_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that deletes a user account.
+ *
+ * Sample:
+ * {
+ *   "query": {
+ *      "id": 1234
+ *   }
+ * }
+ */
+class UserResetPasswordCommand extends Command {
+	/**
+	 * @var integer The id of the user to delete.
+	 */
+	private $user_id_reset;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {
+		$this->user_id_reset = (int)$this->query( 'id', null );
+		if( $this->user_id_reset <= 0 ) {
+			throw new ClientException( 'Invalid user id', ERROR_INVALID_FIELD_VALUE, array( 'id' ) );
+		}
+
+		# Ensure user has access level to delete users
+		if( !access_has_global_level( config_get_global( 'manage_user_threshold' ) ) ) {
+			throw new ClientException( 'Access denied to reset user password', ERROR_ACCESS_DENIED );
+		}
+
+		$t_user = user_get_row( $this->user_id_reset );
```

Comment:
I assume you tested resetting users that don't exist.  I can see the user_get_row() throwing so it should work, but test to make sure all is working as expected.